### DISCUSSION TITLE
Primeng Table -> Clear the filters which are not applied when filter menu closed by clicking outside the menu

### DISCUSSION THREAD

**Discussion by gopikrishna-manifest** [primefaces/community#1718] [PrimeNG] [answered]
Hi Team,
We are using Primeng Tables filter option. We need to clear the filters which are selected but not applied when we close filter menu by clicking outside of the menu. 
**Steps to reproduce:**
1) Open the Primeng filter menu example. [https://primeng.org/table#filter-menu](https://primeng.org/table#filter-menu)
2) Click on Filter icon beside Name / Country column and open the filter menu.
3) Enter some text in the text box 
![Screenshot_4](https://github.com/primefaces/community/assets/85510959/4ecf0c6b-eba1-4217-8a39-45836e8d1477)
4) Click out side the filter menu. Now the filter menu will close without applying the filter
5) But you can see the filter icon turns to blue indicating that some filter applied to that column
![image](https://github.com/primefaces/community/assets/85510959/08e1944b-0971-4ebe-81af-92419a19e917)

6) Again open the Filter menu for the same column
7) You can see the same text in the filter.
![image](https://github.com/primefaces/community/assets/85510959/a4a2f8a4-51ad-4266-937e-f70c09bba187)

**Expected Result**
1) It should clear the filters which are not applied 
2) The Filter icon shouldn't turn to blue

Actual Result
1) It is not clearing the filters which are not applied when we click outside the menu
2) The icon turning to blue even no filter applied to that column.

Is there any workaround for this issue?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, clearing the input field is not necessary by default and users should do it with `filterTemplate` option of the column filter element. You can check the primeng.org/table#filter-menu for the use of templating. However the button should only be highlighted when a filter is applied. Could you please open an issue for that so we can address it in the upcoming releases?

**Reply by Vlad66M** [reply]
Can you give more info on how to do that with filterTemplate?
